cpp/include/gar/utils/status.h

```diff
@@ -69,9 +91,6 @@ enum class StatusCode : unsigned char {
   kKeyError,
   kTypeError,
   kInvalid,
-  kInvalidValue,
```

Comment:
Could you please add some comments for the enum class to briefly introduce the types of different status code?

Reply:
like this: https://arrow.apache.org/docs/cpp/api/support.html#error-return-and-reporting